src/main/kotlin/io/curity/dynamoDBDataAccessProvider/DynamoDBClient.kt

```diff
@@ -51,51 +51,99 @@ import software.amazon.awssdk.services.sts.StsClient
 import software.amazon.awssdk.services.sts.model.AssumeRoleRequest
 import software.amazon.awssdk.services.sts.model.AssumeRoleResponse
 import software.amazon.awssdk.services.sts.model.Credentials
+import java.net.URI
+import java.util.*
 
-class DynamoDBClient(private val config: DynamoDBDataAccessProviderDataAccessProviderConfig): ManagedObject<DynamoDBDataAccessProviderDataAccessProviderConfig>(config)
-{
+class DynamoDBClient(private val config: DynamoDBDataAccessProviderDataAccessProviderConfig) : ManagedObject<DynamoDBDataAccessProviderDataAccessProviderConfig>(config) {
     private val _awsRegion = Region.of(config.getAwsRegion().awsRegion)
-    private val client: DynamoDbClient = initializeDynamoDBClient()
+    private val client: Optional<DynamoDbClient> = initializeDynamoDBClient()
 
-    private fun initializeDynamoDBClient(): DynamoDbClient
-    {
+    private fun initializeDynamoDBClient(): Optional<DynamoDbClient> {
         val accessMethod = config.getDynamodbAccessMethod()
         /*Use Instance Profile from IAM Role applied to EC2 instance*/
-        var creds: AwsCredentialsProvider?
+        var creds: Optional<AwsCredentialsProvider>
 
-        if (accessMethod.isEC2InstanceProfile.isPresent && accessMethod.isEC2InstanceProfile.get()) {
-            creds = InstanceProfileCredentialsProvider.builder().build()
-        } else if (accessMethod.accessKeyIdAndSecret.isPresent) {
-            val keyIdAndSecret = accessMethod.accessKeyIdAndSecret.get()
-            creds = StaticCredentialsProvider.create(AwsBasicCredentials.create(keyIdAndSecret.accessKeyId.get(), keyIdAndSecret.accessKeySecret.get()))
-
-            /* roleARN is present, get temporary credentials through AssumeRole */
-            if (keyIdAndSecret.awsRoleARN.isPresent) {
-                creds = getNewCredentialsFromAssumeRole(creds, keyIdAndSecret.awsRoleARN.get())
+        logger.info("AWS Direct DbClient?: {}", accessMethod.aWSDirect.isPresent)
+        logger.info("AWS EC2Instance?: {}", accessMethod.isEC2InstanceProfile.isPresent)
+        logger.info("AWS awsProfile?: {}", accessMethod.aWSProfile.isPresent)
+        try {
+            if (accessMethod.aWSDirect.isPresent) {
+                logger.info("AWS Direct DbClient?: hostname {}, ", accessMethod.aWSDirect.get().hostname)
```

Comment:
Can you please remove some of the `logger` methods? At least the infos here and above and the `logger.warn("above ")`. Also, maybe it's better to change those loggers to debug? (At least I think it's better not to use `warn` for informative reasons).